packages/manager/src/features/CloudPulse/Utils/ReusableDashboardFilterUtils.ts

```diff
@@ -0,0 +1,165 @@
+import { FILTER_CONFIG } from './FilterConfig';
+
+import type { DashboardProperties } from '../Dashboard/CloudPulseDashboard';
+import type { FilterValueType } from '../Dashboard/CloudPulseDashboardLanding';
+import type { CloudPulseMetricsAdditionalFilters } from '../Widget/CloudPulseWidget';
+import type { Dashboard, TimeDuration } from '@linode/api-v4';
+
+/**
+ * This interface is used to get method parameters for this utility
+ */
+interface ReusableDashboardFilterUtilProps {
+  /**
+   * The selected dashboard object
+   */
+  dashboardObj: Dashboard;
+  /**
+   * The selected filter values
+   */
+  filterValue: { [key: string]: FilterValueType };
+  /**
+   * The selected resource id
+   */
+  resource: number;
+  /**
+   * The selected time duration
+   */
+  timeDuration?: TimeDuration;
+}
+
+/**
+ * @param props The props required for constructing the dashboard properties
+ * @returns The properties compatible for rendering dashboard component
+ */
+export const getDashboardProperties = (
+  props: ReusableDashboardFilterUtilProps
+): DashboardProperties => {
+  const { dashboardObj, filterValue, resource, timeDuration } = props;
+  const timeDurationObj: TimeDuration = {
+    unit: 'min',
+    value: 30,
+  };
+  return {
+    additionalFilters: constructDimensionFilters({
+      dashboardObj,
+      filterValue,
+      resource,
+    }),
+    dashboardId: dashboardObj.id,
+    duration: timeDuration ?? timeDurationObj,
+    resources: [String(resource)],
+    savePref: false,
+  };
```

Comment:
```suggestion
  return {
    additionalFilters: constructDimensionFilters({
      dashboardObj,
      filterValue,
      resource,
    }),
    dashboardId: dashboardObj.id,
    duration: timeDuration ??  { unit: 'min', value: 30 },
    resources: [String(resource)],
    savePref: false,
  };
```
we can directly pass the default value instead of storing in the object